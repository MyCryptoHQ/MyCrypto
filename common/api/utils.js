export function checkHttpStatus(response) {
  if (response.status >= 200 && response.status < 300) {
    return response;
  } else {
    let error = new Error(response.statusText);
    error.response = response;
    throw error;
  }
}

export function parseJSON(response) {
  return response.json();
}
