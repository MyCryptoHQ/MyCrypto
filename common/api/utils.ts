export function checkHttpStatus(response) {
  if (response.status >= 200 && response.status < 300) {
    return response;
  } else {
    return new Error(response.statusText);
  }
}

export function parseJSON(response) {
  return response.json();
}

export async function handleJSONResponse(response, errorMessage) {
  if (response.ok) {
    return await response.json();
  }
  if (errorMessage) {
    throw new Error(errorMessage);
  }
  return false;
}
