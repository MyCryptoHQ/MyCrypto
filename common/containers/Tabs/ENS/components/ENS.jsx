// @flow
import * as React from 'react';
import Title from './Title';
import GeneralInfoPanel from './GeneralInfoPanel';
import UnfinishedBanner from './UnfinishedBanner';
import NameInput from './NameInput';
import NameResolve from './NameResolve';

type ContainerTabPaneActiveProps = {
  children: React.Element<any>
};

const ContainerTabPaneActive = ({ children }: ContainerTabPaneActiveProps) =>
  <section className="container">
    <div className="tab-content">
      <main className="tab-pane active">
        <section role="main" className="row">
          {children}
        </section>
      </main>
    </div>
  </section>;

type Props = {
  ensState: {},
  resolveDomainRequested: (domain: string) => void
};
const ENS = (props: Props) => {
  return (
    <ContainerTabPaneActive>
      <UnfinishedBanner />
      <Title />
      <NameInput resolveDomainRequested={props.resolveDomainRequested} />
      <NameResolve {...props.ensState} />
      <GeneralInfoPanel />
    </ContainerTabPaneActive>
  );
};

export default ENS;
