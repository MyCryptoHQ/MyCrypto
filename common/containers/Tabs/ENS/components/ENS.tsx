import * as React from 'react';
import Title from './Title';
import { GeneralInfoPanel } from './GeneralInfoPanel';
// import UnfinishedBanner from './UnfinishedBanner';
import NameInput from './NameInput';
import { NameResolve } from './NameResolve';
import { AppState } from 'reducers';
import TabSection from 'containers/TabSection';

interface Props {
  ensState: AppState['ens'];
  resolveDomainRequested(domain: string): void;
}

const ENS = (props: Props) => {
  return (
    <TabSection>
      {/* <UnfinishedBanner /> */}
      <Title />
      <NameInput resolveDomainRequested={props.resolveDomainRequested} />
      <NameResolve {...props.ensState} />
      <GeneralInfoPanel />
    </TabSection>
  );
};

export default ENS;
