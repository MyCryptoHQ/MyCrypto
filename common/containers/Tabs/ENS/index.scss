@import 'common/sass/variables';

.ENS {
  &-title {
    text-align: center;
  }

  &-description {
    max-width: 800px;
    margin: 0 auto $space;
  }
}
