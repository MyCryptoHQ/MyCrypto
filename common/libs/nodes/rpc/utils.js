// @flow
// Ref: https://github.com/ethereum/wiki/wiki/JSON-RPC

import type Big from 'big.js';
import { toHex } from 'libs/values';
import { toBuffer } from 'ethereumjs-util';

// When encoding QUANTITIES (integers, numbers): encode as hex, prefix with "0x", the most compact representation (slight exception: zero should be represented as "0x0").
export function encodeQuantity(value: Big): string {
  return '0x' + (toHex(value) || '0');
}

// When encoding UNFORMATTED DATA (byte arrays, account addresses, hashes, bytecode arrays): encode as hex, prefix with "0x", two hex digits per byte.
export function encodeData(value: string | Buffer): string {
  return '0x' + toBuffer(value).toString('hex');
}
