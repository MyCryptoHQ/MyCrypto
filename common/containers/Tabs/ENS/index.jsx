// @flow
import { connect } from 'react-redux';
import ENS from './components/ENS';
import type { State } from 'reducers';
import { resolveDomainRequested } from 'actions/ens';

const mapStateToProps = (state: State) => {
  return { ensState: state.ens };
};

export default connect(mapStateToProps, {
  resolveDomainRequested
})(ENS);
