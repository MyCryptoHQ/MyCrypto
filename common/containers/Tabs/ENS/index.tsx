import React from 'react';
import { NameInput, NameResolve } from './components';
import TabSection from 'containers/TabSection';
import { NewTabLink } from 'components/ui';
import { donationAddressMap } from 'config';
import './index.scss';

const ENSDocsLink = () => (
  <NewTabLink
    href="https://ens.readthedocs.io/en/latest/introduction.html"
    content="Ethereum Name Service"
  />
);

export default class ENSClass extends React.Component<{}> {
  public render() {
    return (
      <TabSection isUnavailableOffline={true}>
        <div className="Tab-content">
          <section className="Tab-content-pane">
            <div className="ENS">
              <h1 className="ENS-title">Ethereum Name Service</h1>
              <p className="ENS-description">
                The <ENSDocsLink /> is a distributed, open, and extensible naming system based on
                the Ethereum blockchain. Once you have a name, you can tell your friends to send ETH
                to <code>ensdomain.eth</code> instead of
                <code>{donationAddressMap.ETH.substr(0, 12)}...</code>
              </p>

              <NameInput />
            </div>
          </section>

          <NameResolve />
        </div>
      </TabSection>
    );
  }
}
