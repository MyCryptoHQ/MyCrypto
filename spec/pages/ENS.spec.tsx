import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ENS from 'containers/Tabs/ENS';
import shallowWithStore from '../utils/shallowWithStore';
import { createMockStore } from 'redux-test-utils';
import { createMockRouteComponentProps } from '../utils/mockRouteComponentProps';
import { RouteComponentProps } from 'react-router';

Enzyme.configure({ adapter: new Adapter() });

const routeProps: RouteComponentProps<any> = createMockRouteComponentProps({
  match: { path: '/ens', url: '/ens', isExact: false, params: {} },
  location: { pathname: '/ens', search: '', hash: '', key: 'e08jz7' },
  history: {
    length: 2,
    action: 'PUSH',
    location: { pathname: '/ens', search: '', hash: '', key: 'e08jz7', state: {} }
  }
});

describe('snapshot test', () => {
  it('ENS component', () => {
    const testState = {};
    const store = createMockStore(testState);
    const component = shallowWithStore(<ENS {...routeProps} />, store);

    expect(component).toMatchSnapshot();
  });
});
