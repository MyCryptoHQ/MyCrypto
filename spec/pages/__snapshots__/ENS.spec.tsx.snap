// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshot 1`] = `
<ENS
  resolveDomainRequested={[Function]}
/>
`;
